harvester_e2e_tests/integration/test_hosts.py

```diff
@@ -123,68 +124,77 @@ def vm_force_reset_policy(api_client):
-    """
-    Test the hosts are the nodes which make the cluster
-    Covers:
-        hosts-01-Negative test-Verify the state for Powered down node
-    """
-    assert 1 < len(available_node_names), (
-        f"Cluster has {len(available_node_names)} available node, not enough for poweroff test."
-    )
-
-    _, nodes_info = api_client.hosts.get()
-    node = nodes_info['data'][-1]
-    node_ip = next(val["address"] for val in node['status']['addresses']
-                   if val["type"] == "InternalIP")
-
-    rc, out, err = host_state.power(node['id'], node_ip, on=False)
-    assert rc == 0, (f"Failed to PowerOff node {node['id']} with error({rc}):\n"
-                     f"stdout: {out}\n\nstderr: {err}")
-    sleep(host_state.delay)  # Wait for the node to disappear
-    endtime = datetime.now() + timedelta(seconds=wait_timeout)
-    while endtime > datetime.now():
-        _, metric = api_client.hosts.get_metrics(node['id'])
-        if metric.get("status") == 404:
-            break
-        sleep(5)
-    else:
-        raise AssertionError(f"Node {node['id']} still available after PowerOff script executed"
-                             f", script path: {host_state.path}")
-
-    _, node = api_client.hosts.get(node['id'])
-    assert node["metadata"]["state"]["name"] in ("in-progress", "unavailable")
+class TestHostState:
+    @pytest.mark.dependency(name="host_poweroff")
+    def test_poweroff_state(api_client, host_state, wait_timeout, available_node_names):
+        """
+        Test the hosts are the nodes which make the cluster
+        Covers:
+            hosts-01-Negative test-Verify the state for Powered down node
+        """
+        assert 2 <= len(available_node_names), (
+            f"The cluster only have {len(available_node_names)} available node."
+            " It's not enough for power off test."
+        )
+        _, node = api_client.hosts.get(available_node_names[-1])
+
+        node_ip = next(val["address"] for val in node['status']['addresses']
+                       if val["type"] == "InternalIP")
+
+        rc, out, err = host_state.power(node['id'], node_ip, on=False)
+        assert rc == 0, (f"Failed to PowerOff node {node['id']} with error({rc}):\n"
+                         f"stdout: {out}\n\nstderr: {err}")
+        sleep(host_state.delay)  # Wait for the node to disappear
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, metric = api_client.hosts.get_metrics(node['id'])
+            if 404 == code:
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Node {node['id']} still available after PowerOff script executed"
+                f", script path: {host_state.path}"
+            )
 
+        _, node = api_client.hosts.get(node['id'])
+        assert node["metadata"]["state"]["name"] in ("in-progress", "unavailable")
```

Comment:
Should we expect "in-progress" after power-off?

Reply:
When a host is powered off, it's metric will be a real time check and return `404` as we just checked.
But metadata of state for the host will need to sync up (by k8s), so `In-progress` is one of the possible state in it.